tests/unit/package_managers/bundler/test_parser.py

```diff
@@ -215,6 +226,62 @@ def test_dependencies_could_be_downloaded(
     mock_downloader.assert_called_once_with(expected_source_url, expected_destination)
 
 
+@mock.patch("cachi2.core.package_managers.bundler.parser.download_binary_file")
+def test_binary_gem_dependencies_could_be_downloaded(
+    mock_downloader: mock.MagicMock,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    base_destination = RootedPath("/tmp/foo")
+    source = "https://rubygems.org"
+    platform = "m6502_wm"
+    dependency = GemPlatformSpecificDependency(
+        name="foo",
+        version="0.0.2",
+        source=source,
+        platform=platform,
+    )
+    expected_source_url = f"{source}/downloads/foo-0.0.2-{platform}.gem"
+    expected_destination = base_destination.join_within_root(Path(f"foo-0.0.2-{platform}.gem"))
+
+    dependency.download_to(base_destination)
+
+    assert relaxed_in("Downloading platform-specific gem", caplog.messages)
+    mock_downloader.assert_called_once_with(expected_source_url, expected_destination)
+
+
+@mock.patch("cachi2.core.package_managers.bundler.parser.download_binary_file")
+def test_binary_gem_dependencies_could_be_downloaded_for_damaged_platforms(
+    mock_downloader: mock.MagicMock,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    base_destination = RootedPath("/tmp/foo")
+    source = "https://rubygems.org"
+    platform = "aarch-linux"
+    dependency = GemPlatformSpecificDependency(
+        name="foo",
+        version="0.0.2",
+        source=source,
+        platform=platform,
+    )
+    expected_source_url = f"{source}/downloads/foo-0.0.2-{platform}.gem"
+    expected_corrected_source_url = f"{source}/downloads/foo-0.0.2-{platform}-gnu.gem"
+    expected_destination = base_destination.join_within_root(Path(f"foo-0.0.2-{platform}.gem"))
+    expected_corrected_destination = base_destination.join_within_root(
+        Path(f"foo-0.0.2-{platform}-gnu.gem")
+    )
+    expected_calls = [
+        mock.call(expected_source_url, expected_destination),
+        mock.call(expected_corrected_source_url, expected_corrected_destination),
+    ]
+
+    mock_downloader.side_effect = [FetchError(reason="testing"), "ok"]
+
+    dependency.download_to(base_destination)
+
+    assert relaxed_in("Downloading platform-specific gem", caplog.messages)
+    mock_downloader.assert_has_calls(expected_calls)
+
+
```

Comment:
In context of this review, this test case should go away.

Reply:
Do you mean `test_binary_gem_dependencies_could_be_downloaded_for_damaged_platforms`? If yes then I disagree: it verifies that our code handles externally produced damaged data so it must stay.